charmcraft/commands/store/client.py

```diff
@@ -46,9 +47,10 @@
 
 def build_user_agent():
     """Build the charmcraft's user agent."""
-    # XXX Facundo 2020-06-29: we need to include here at least the platform, maybe
-    # architecture, etc. Related: issue #74.
-    return "charmcraft/{}".format(__version__)
+    return "charmcraft/{} ({}; {}) python/{}".format(__version__,
```

Comment:
Just for reference, this is the UA we agreed to for Snapcraft
`"snapcraft/{snapcraft.__version__} {testing}{os_platform} ({arch})"`

Which can lead to:

- snapcraft/4.4 Unknown/Unknown Version (armhf)
- snapcraft/4.4 Ubuntu/18.04 (armhf)
- snapcraft/4.4 (testing) Ubuntu/18.04 (armhf)
- snapcraft/4.5 Windows (amd64)
- snapcraft/4.5 Darwin (amd64)

Reply:
Ah, I see, much more detailed than my simple implementation. I am more than fine with fleshing this out, I should have started a discussion around this anyway but wanted to jump into something ( :smile: ). Standardizing with Snapcraft seems like the best idea to me since it has been discussed and agreed on there.

If anyone has any thoughts please let me know! cc @facundobatista @cjp256

Reply:
Thanks @jbpratt for working on this!